drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -62,6 +62,21 @@
 #define PLLSRC_FREQ 0
 #endif
 
+#define VCO_FREQ	(PLLSRC_FREQ / CONFIG_CLOCK_STM32_PLL_M_DIVISOR)
+
+#if (1000000UL <= VCO_FREQ && VCO_FREQ <= 2000000UL)
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_1_2
+#elif (2000000UL < VCO_FREQ && VCO_FREQ <= 4000000UL)
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_2_4
+#elif (4000000UL < VCO_FREQ && VCO_FREQ <= 8000000UL)
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_4_8
+#elif (8000000UL < VCO_FREQ && VCO_FREQ <= 16000000UL)
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_8_16
+#else
+#pragma message STRINGIFY(VCO_FREQ)
```

Comment:
Tryong to get into the error, I got:
```
/local/mcu/zephyrproject/zephyr/drivers/clock_control/clock_stm32_ll_h7.c:76:9: note: #pragma message: (8000000 / 63)
   76 | #pragma message STRINGIFY(VCO_FREQ)
      |         ^~~~~~~
/local/mcu/zephyrproject/zephyr/drivers/clock_control/clock_stm32_ll_h7.c:77:2: error: #error "PLL1 VCO frequency input range out of range " STRINGIFY(VCO_FREQ)
   77 | #error "PLL1 VCO frequency input range out of range " STRINGIFY(VCO_FREQ)
      |  ^~~~~
```
I guess this is not the expected result...

Reply:
Damn, sorry, I'm doing to many things at once, this should not be there
BTW is there a way how to print out result of macro inside `#pragma` or `#error`

Reply:
@xhpohanka There is no way, preprocessor doesn't compute the result, it's the compiler's job. But you can still do math in #if statements.